rest/api_test.go

```diff
@@ -7918,3 +7918,722 @@ func TestDocChannelSetPruning(t *testing.T) {
+		// Create a document.
+		docID := "qux"
+		revID := rt.createDoc(t, docID)
+		require.Equal(t, "1-45ca73d819d5b1c9b8eea95290e79004", revID)
+
+		// Add an attachment to the document.
+		attName := "qux.txt"
+		attBody := "this is the body of attachment qux.txt"
+		revID = storeAttachment(docID, revID, attName, attBody)
+		require.Equal(t, "2-abe7339f42c9218acb7b906f5977adcf", revID)
+
+		// Retrieve the attachment added from the document.
+		actualAttBody := retrieveAttachment(docID, attName)
+		require.Equal(t, attBody, actualAttBody)
+
+		// Get the doc and check the attachment meta.
+		attachments := retrieveAttachmentMeta(docID)
+		require.Len(t, attachments, 1)
+		meta, ok := attachments[attName].(map[string]interface{})
+		require.True(t, ok)
+		assert.True(t, meta["stub"].(bool))
+		assert.Equal(t, "content/type", meta["content_type"].(string))
+		assert.Equal(t, "sha1-0naD6SgfLVDr+zakP8RkNlBYORw=", meta["digest"].(string))
+		assert.Equal(t, float64(38), meta["length"].(float64))
+		assert.Equal(t, float64(2), meta["revpos"].(float64))
+		assert.Equal(t, float64(2), meta["ver"].(float64))
+
+		// Retrieve attachment key used for internal attachment storage and retrieval.
+		attKeyOld := retrieveAttachmentKey(docID, attName)
+		require.Equal(t, "_sync:att2:IfWNJ/gn0pX/zYYMZQRWheO68a1FBsqgFAETsxZkdTQ=:sha1-0naD6SgfLVDr+zakP8RkNlBYORw=", attKeyOld)
+
+		// Update the attachment body bytes.
+		attBodyUpdated := "this is the updated body of attachment qux.txt"
+		revID = storeAttachment(docID, revID, attName, attBodyUpdated)
+		require.Equal(t, "3-9d8d0bf2e87982d97356a181a693291b", revID)
+
+		// Retrieve the updated attachment added from the document.
+		actualAttBody = retrieveAttachment(docID, attName)
+		require.Equal(t, attBodyUpdated, actualAttBody)
+
+		// Get the document and check the attachment metadata.
+		attachments = retrieveAttachmentMeta(docID)
+		require.Len(t, attachments, 1)
+		meta, ok = attachments[attName].(map[string]interface{})
+		require.True(t, ok)
+		assert.True(t, meta["stub"].(bool))
+		assert.Equal(t, "content/type", meta["content_type"].(string))
+		assert.Equal(t, "sha1-dDdppdY7RC4gq550G7eGJgQmk6g=", meta["digest"].(string))
+		assert.Equal(t, float64(46), meta["length"].(float64))
+		assert.Equal(t, float64(3), meta["revpos"].(float64))
+		assert.Equal(t, float64(2), meta["ver"].(float64))
+
+		// Check whether the old attachment blob is removed from the underlying storage.
+		rt.requireDocNotFound(attKeyOld)
+
+		// Retrieve new attachment key used for internal attachment storage and retrieval.
+		attKeyNew := retrieveAttachmentKey(docID, attName)
+		require.Equal(t, "_sync:att2:IfWNJ/gn0pX/zYYMZQRWheO68a1FBsqgFAETsxZkdTQ=:sha1-0naD6SgfLVDr+zakP8RkNlBYORw=", attKeyOld)
+		require.NotEqual(t, attKeyOld, attKeyNew)
+
+		// Perform cleanup after the test ends.
+		rt.purgeDoc(docID)
+	})
+
+	t.Run("multiple attachments removal upon document update", func(t *testing.T) {
```

Comment:
Do you have test coverage for 'document with multiple attachments, and only one is removed'?

Reply:
Included a new test "doc with multiple attachments and removal of a single one upon document update" that covers the case of document with multiple attachments, and only one is removed.